rllib/algorithms/sac/sac.py

```diff
@@ -371,3 +430,133 @@ def get_default_policy_class(
             return SACTorchPolicy
         else:
             return SACTFPolicy
+
+    @override(DQN)
+    def training_step(self) -> ResultDict:
+        # Check, if we use `RolloutWorker` or `EnvRunner`.
+        use_rollout_worker = self.config.env_runner_cls is None or issubclass(
+            self.config.env_runner_cls, RolloutWorker
+        )
+
+        # If `RolloutWorker` is used and the old stack, then use as before
+        # the training step of the DQN algorithm.
+        if use_rollout_worker or not self.config._enable_new_api_stack:
```

Comment:
replace w/:
```
if self.config.uses_new_env_runners:
 ...
```

Reply:
Just for clarification: `uses_new_env_runners` can be used here, as the `SingleAgentEnvRunner` s expect an `RLModule`. What, if someone uses the `EnvRunner` as a base class, but not the `RLModule`?

Reply:
Correct. I actually haven't thought about this combination (EnvRunner w/o RLModule). But I don't think we should at this point. if EnvRunner AND NOT RolloutWorker -> then there must be an RLModule (always under property: `env_runner.module`).

Reply:
But you have a point: EnvRunner is the base class.
I guess `self.uses_new_env_runners` should only be True if we are using a sub-class of SingleAgentEnvRunner OR MultiAgentEnvRunner.
Hmm, this hybrid stack is getting tough to maintain :p

Reply:
I think we should clearly point out that the new stack contains 'RLModule' , 'Learner' and 'AgentEnvRunner's. And that hybrid combinations are not covered. So either you do Policy/RolloutWorker or RLModule/EnvRunner. That makes it cleaner and makes maintenance easier imo

Reply:
Sure, but then you have some users already (more or less deeply) invested in this hybrid stack (RLModule + Learner, BUT no EnvRunner), so you don't want to cold-force them to convert their Connectors/Policy/etc.. stack to the new API yet.

We can only deprecate the hybrid stack (at least for PPO) once we fully support multi-agent on it (a few days away).